Templates/DefaultProject/Template/cmake/EngineFinder.cmake

```diff
@@ -13,66 +13,70 @@
-
-set(ENGINE_FINDER_VERBOSE FALSE CACHE BOOL "Increase EngineFinder.cmake verbosity.  Useful for debugging engine selection based on compatibility.")
-
-# Read the engine name from the shared or user project.json
-file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json project_json)
+# Read the 'engine' field in project.json for compatibility with older engines
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json o3de_project_json)
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/project.json)
-
-string(JSON O3DE_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${project_json} engine)
+string(JSON O3DE_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${o3de_project_json} engine)
 if(json_error)
     message(FATAL_ERROR "Unable to read key 'engine' from 'project.json'\nError: ${json_error}")
 endif()
- 
-# Let the user override the engine to use
-if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json)
-    file(READ ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json user_project_json)
-    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json)
-    string(JSON user_project_engine_name ERROR_VARIABLE json_error GET ${user_project_json} engine)
+
+# Allow the user to override the 'engine' field locally using <project>/user/project.json
+cmake_path(SET O3DE_USER_PROJECT_JSON_PATH ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json)
+if(EXISTS "${O3DE_USER_PROJECT_JSON_PATH}")
+    file(READ "${O3DE_USER_PROJECT_JSON_PATH}" user_project_json)
+    set_property(GLOBAL PROPERTY O3DE_USER_PROJECT_JSON ${user_project_json})
+    string(JSON user_project_engine ERROR_VARIABLE json_error GET ${user_project_json} engine)
     if(NOT json_error)
-        set(O3DE_ENGINE_NAME_TO_USE ${user_project_engine_name})
+        # Add cmake configure dependency on user/project.json in case the engine overrides change 
+        set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${O3DE_USER_PROJECT_JSON_PATH}")
+        if(user_project_engine)
+            # Update the project json data so it is available in the O3DE_PROJECT_JSON global property
+            string(JSON o3de_project_json SET "${o3de_project_json}" engine "\"${user_project_engine}\"" )
+            set(project_engine ${O3DE_ENGINE_NAME_TO_USE})
+            set(O3DE_ENGINE_NAME_TO_USE ${user_project_engine})
+        endif()
     endif()
-    string(JSON user_engine_path ERROR_VARIABLE json_error GET ${user_project_json} "engine_path")
 endif()
 
-# Extract optional version specifier in the engine name 
-o3de_get_version_specifier_parts("${O3DE_ENGINE_NAME_TO_USE}" O3DE_ENGINE_NAME_TO_USE O3DE_ENGINE_SPECIFIER_OP O3DE_ENGINE_SPECIFIER_VERSION)
+# Save the project json data with overrides in a global property to avoid performance hit
+# from loading the file multiple times
+set_property(GLOBAL PROPERTY O3DE_PROJECT_JSON ${o3de_project_json})
 
-# for backward compatibility
+# For backwards compatibility
```

Comment:
The backwards compatibility isn't needed in the project templates since this is for a new project